deploy/helm/templates/deployment.yaml

```diff
@@ -0,0 +1,107 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "starboard-operator.fullname" . }}
+  labels:
+    {{- include "starboard-operator.labels" . | nindent 4 }}
+spec:
+  # NOTE: replicas and strategy.type is set like this explicitly to avoid ending
+  #       up with starboard-operator instances disrupting each other. If
+  #       starboard-operator is confirmed to be highly available,
+  #       strategy.type==Recreate can be removed and replicas can be allowed to
+  #       be configurable.
+  replicas: 1
+  strategy:
+    type: Recreate
+  selector:
+    matchLabels:
+      {{- include "starboard-operator.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      annotations:
+        checksum/secret: {{ .Values.envSecret.stringData | toYaml | sha256sum }}
+        {{- with .Values.podAnnotations }}
+        {{- . | toYaml | nindent 8 }}
+        {{- end }}
+      labels:
+        {{- include "starboard-operator.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - name: "{{ .Chart.Name }}"
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          env:
+            - name: OPERATOR_NAMESPACE
+              value: {{ .Release.Namespace | quote }}
+            - name: OPERATOR_TARGET_NAMESPACES
+              value: {{ tpl .Values.targetNamespaces . | quote }}
+            - name: OPERATOR_METRICS_BIND_ADDRESS
+              value: ":8080"
```

Comment:
We could use: {{ print ":" .Values.image.metricsPort | quote }}

Reply:
I think we can avoid needing to add configuration of this because users will only access this through the k8s Service, which in turn points to the Pod's port named `metrics`.

Hmmm, the k8s Service port can currently be configured with `service.port`, but I think we should name that `metricsPort` like you suggest here as that probably makes it less confusing about what the service is meant for currently, and also in case we want to expose something else that isn't metrics on another port.

Reply:
Users of another Helm chat I've worked with have been fine without configuring the container/pod port for a very long time, so we opted there to not add it at any point in time: https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/master/jupyterhub/templates/hub/deployment.yaml#L199-L201

Reply:
I added 9143ca1 to rename service.port to service.metricsPort - do you agree this is sufficient?